src/mapGenerator/mapgenProcess.py

```diff
@@ -0,0 +1,73 @@
+from decorators import with_logger
+from PyQt5.QtCore import QProcess, QProcessEnvironment, QEventLoop, Qt
+from PyQt5.QtWidgets import QMessageBox, QApplication, QProgressDialog
+import os
+import fafpath
+
+@with_logger
+class MapGeneratorProcess(object):
+    def __init__(self, gen_path, out_path, seed, version, mapName):
+        self._progress = QProgressDialog()
+        self._progress.setWindowTitle("Generating map, please wait...")
+        self._progress.setCancelButtonText("Cancel")
+        self._progress.setWindowFlags(Qt.CustomizeWindowHint | Qt.WindowTitleHint)
+        self._progress.setAutoReset(False)
+        self._progress.setModal(1)
+        self._progress.setMinimum(0)
+        self._progress.setMaximum(250)
+        self._progress.canceled.connect(self.close)
+        self.progressCounter = 1
+        
+        self.map_generator_process = QProcess()
+        self.map_generator_process.readyReadStandardOutput.connect(self.on_log_ready)
+        self.map_generator_process.readyReadStandardError.connect(self.on_error_ready)
+        self.map_generator_process.finished.connect(self.on_exit)
+        
+        self.java_path = os.path.join(fafpath.get_libdir(), "ice-adapter", "jre", "bin", "java.exe")
```

Comment:
duplicated code, provide a unified function shared with the adapter code for obtaining the jre location.

Reply:
hmm, currently it launches ice via faf-ice-adapter.exe from this archive https://github.com/FAForever/java-ice-adapter/releases/download/v1.0.0/faf-ice-adapter-jre-base.7z

anyway, I'll make a function in utils